codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/Waiters2.java

```diff
@@ -0,0 +1,763 @@
+                writer.write(
+                        "if err != nil { return false, "
+                                + "fmt.Errorf(\"error parsing boolean from string %w\", err)}");
+
+                writer.openBlock("if $L == bv {", "}", searchResult.ident(), () -> {
+                    writeMatchedAcceptorReturn(writer, acceptor);
+                });
+                break;
+
+            case ALL_STRING_EQUALS:
+                writer.write("match := len($L) > 0", searchResult.ident());
+                writer.openBlock("for _, v := range $L {", "}", searchResult.ident(), () -> {
+                    writer.write("""
+                        if string(v) != expectedValue {
+                            match = false
+                            break
+                        }""");
+                });
+                writer.write("");
+
+                writer.openBlock("if match {", "}", () -> {
+                    writeMatchedAcceptorReturn(writer, acceptor);
+                });
+                break;
+
+            case ANY_STRING_EQUALS:
+                writer.write("var match bool");
+                writer.openBlock("for _, v := range $L {", "}", searchResult.ident(), () -> {
+                    writer.write("""
+                        if string(v) == expectedValue {
+                            match = true
+                            break
+                        }""");
+                });
+                writer.write("");
+
+                writer.openBlock("if match {", "}", () -> {
+                    writeMatchedAcceptorReturn(writer, acceptor);
+                });
+                break;
+
+            default:
+                throw new CodegenException(
+                        String.format("Found unknown waiter path comparator, %s", comparator.toString()));
+        }
+    }
+
+
+    /**
+     * Writes return statement for state where a waiter's acceptor state is a match.
+     *
+     * @param writer   the Go writer
+     * @param acceptor the waiter acceptor who's state is used to write an appropriate return statement.
+     */
+    private void writeMatchedAcceptorReturn(GoWriter writer, Acceptor acceptor) {
+        switch (acceptor.getState()) {
+            case SUCCESS:
+                writer.write("return false, nil");
+                break;
+
+            case FAILURE:
+                writer.addUseImports(SmithyGoDependency.FMT);
+                writer.write("return false, fmt.Errorf(\"waiter state transitioned to Failure\")");
+                break;
+
+            case RETRY:
+                writer.write("return true, nil");
+                break;
+
+            default:
+                throw new CodegenException("unknown acceptor state defined for the waiter");
+        }
+    }
+
+    private String generateWaiterOptionsName(
+            String waiterName
+    ) {
+        waiterName = StringUtils.capitalize(waiterName);
+        return String.format("%sWaiterOptions", waiterName);
```

Comment:
[tiny nit] Java can be simpler and just do `return waiterName + "WaiterOptions"`